### DISCUSSION TITLE
OTLP Endpoint Setup

### DISCUSSION THREAD

**Discussion by devidw** [huggingface/text-generation-inference#1247] [Q&A]
- https://github.com/huggingface/text-generation-inference#distributed-tracing
- https://huggingface.co/docs/text-generation-inference/basic_tutorials/launcher#otlpendpoint

Wondering what exactly can be passed:

1. A HTTP URL or gRPC URL, or both possible?
2. Does it have to point to a specific endpoint like `/v1/metrics` or just the root OTLP endpoint?
3. How to deal with authentication, many services such as logit.io or honeycomb.io that offer hosted OTLP require passing headers with api key

**Comment by rodrigomasiniai** [comment]
Any updates on this?

**Reply by michalnicp** [reply]
I have discovered that only gRPC is supported. Additionally, passing an empty or http URL will cause it to hang when a request is made.
